### PR TITLE
Incremental backup: accept GTID position without 'MySQL56/' flavor prefix

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

The user may now issue:

```sh
$ vtctlclient -- Backup --incremental_from_pos "16b1039f-22b6-11ed-b765-0a43f95f28a3:1-615" zone1-0000000102
```

And Vitess will infer that they mean `"MySQL56/16b1039f-22b6-11ed-b765-0a43f95f28a3:1-615"`. `MySQL56` is the only flavor supported, anyway. As long as the GTID set is valid and has no parsing errors, the `MySQL56/` can be omitted.

## Related Issue(s)

Fixes https://github.com/vitessio/vitess/issues/13415

Documentation: https://github.com/vitessio/website/pull/1533

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [ ] Tests were added or are not required
-   [ ] Did the new or modified tests pass consistently locally and on the CI
-   [ ] Documentation was added or is not required

### CODE REVIEW

**Comment by shlomi-noach** [comment]
Documentation PR: https://github.com/vitessio/website/pull/1533

**Review by dbussink — APPROVED** [review]

File: go/vt/mysqlctl/builtinbackupengine2_test.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2022 The Vitess Authors.
```
Comment (GuptaManan100) [inline go/vt/mysqlctl/builtinbackupengine2_test.go:2]: The license year has to change.

**Review by GuptaManan100 — APPROVED** [review]
Rest, looks good to me.

File: go/vt/mysqlctl/builtinbackupengine2_test.go (hunk 1 above)
Comment (shlomi-noach) [reply go/vt/mysqlctl/builtinbackupengine2_test.go]: fixed

**Pull request merged**